### PR TITLE
doc: Document seamless upgrade process for remote-node identity change

### PR DESCRIPTION
**Pull request by tgraf**
This PR builds on top of #13038

The upgrade instructions have been incomplete and did not allow for a
seamless upgrade from 1.6.x to 1.7.0.

### CODE REVIEW

**Review by christarazi — APPROVED** [review]

File: Documentation/install/upgrade.rst [hunk 1]
```diff
@@ -427,53 +427,79 @@ New ConfigMap Options
...
+       check and start dropping traffic originating from the host identity
+       received over the network. It is important to make this change
+       independent of enabling the transmission to a avoid a mix of nodes with
+       restrictive receive behavior while old nodes are still transmitting with
+       the legacy host identity.
+
+    4. Network policies must be audited to ensure policies no longer allow on the
+       old behavior of the host entity allowing from all nodes in the cluster.
+       After this step has been completed, remote node identity policy can be
+       enabled by setting ``enable-remote-node-identity: true``.
```
Comment (joestringer) [inline Documentation/install/upgrade.rst:495]: I wonder whether there is a practical difference to splitting these into separate steps or whether they could be combined together? Until step 4 is executed, from a policy perspective, allows "from host" will still allow traffic from remote nodes.

**Review by joestringer — APPROVED** [review]
LGTM. Just one question on whether the last two steps could be combined.

**Comment by joestringer** [comment]
test-docs-please

**Comment by joestringer** [comment]
We can iterate further on top if necessary, merged.

It likely also makes sense for us to revisit forward-porting these docs changes to master / v1.8 given that these instructions are also present there.

**Pull request merged**